src/lib/ollama.ts

```diff
@@ -1,25 +1,46 @@
 import { get } from 'svelte/store';
 import { settingsStore } from '$lib/store';
+import type { ChatRequest, ChatResponse, ListResponse } from 'ollama/browser';
 
-export type OllamaModel = {
-	name: string;
-	model: string;
-	modified_at: string;
-	size: number;
-	digest: string;
-	details: {
-		parent_model: string;
-		format: string;
-		family: string;
-		families: string[] | null;
-		parameter_size: string;
-		quantization_level: string;
-	};
-};
-
-export type OllamaTagResponse = {
-	models: OllamaModel[];
-};
+export async function ollamaChat(
+  payload: ChatRequest, 
+  abortSignal: AbortSignal, 
+  onChunk: (content: string) => void
+) {
```

Comment:
The implementation of `ollamaChat()` is very similar to the old [`ollamaGenerate()`](https://github.com/fmaclen/hollama/pull/125/files#diff-31ae89989234ded3c6f93e8e5fe7cc2c30e93536cc027383faae05989bf79205L45-L55)